library/src/lapack/roclapack_getf2.cpp

```diff
@@ -127,7 +128,7 @@ rocblas_status rocsolver_sgetf2_npvt(rocblas_handle handle,
                                      rocblas_int* info)
 {
     rocblas_int* ipiv = nullptr;
-    return rocsolver_getf2_impl<false, float>(handle, m, n, A, lda, ipiv, info);
+    return rocsolver_getf2_impl<float>(handle, m, n, A, lda, ipiv, info, false);
```

Comment:
There were three main reasons for why I left out the `pivot` argument in the template functions, which I arrived at during the work on getri_npvt:
1. I have an unverified belief that using fewer arguments makes kernel launches faster, and I extended that approach to host code.
2. Because of argument checking, I was fairly confident that `ipiv` should only be a `nullptr` in the npvt case. At the time, I hadn't thought of workspace arrays like `iipiv`, which throws a wrench in that idea.
3. I didn't want to add a `pivot` argument to `getrs`, which handles the pivoting for `getri_outofplace`. For consistency, I didn't add it to other functions either.

The first two reasons are pretty weak, so I'm happy to add the argument in. I guess the only question is if we're okay adding it to getrs as well, even though it does not currently support npvt routines.

Reply:
I see...

I think it is OK. `getrs` actually supports npvt cases!! it just does not need to have explicit npvt APIs; it behaves according to how A was factorized: If A=PLU, then `getrs` apply the permutations, if A=LU then `getrs` does not care about permutations... 
The thing here is how we let `getrs` know that it needs to apply permutations? Right now we use
`pivot = (ipiv != nullptr)` inside `getrs_template`, but we can also pass the already set argument `pivot` to `getrs_template`...

As mentioned before, we know that both approaches will work (they are equivalent as the arg checking won't let us have `pivot = true && ipiv = nullptr`, for example), I just think that we should use only one approach and not combine them. For the reasons I exposed before (and in the `getri_npvt` PR) I feel more comfortable with cascading down the argument `pivot` everywhere it is needed... is like the `shift`, `stride` and `batch_count` arguments, you just set them once in the implementation of the C API, and pass them everywhere so you don't have to take care of them anymore. 

So, in conclusion... I would be happy with either option, combining `pivot` and the `ipiv != nullptr` or only using `pivot`. I leave it up to you :)

Let me know when you are ready for me to merge this PR.

Reply:
I've made the change. Everything is compiling and running correctly (I haven't tested the optimized kernels, though, since they take such a long time to compile).

Looks like merging develop into my branch has caused the diff to include all the changes made in the getri_npvt PR. I'd recommend isolating the "Add pivot argument to template functions" commit for review since that's where my actual changes were made.

Ready to merge when you are.